climada/util/api_client.py

```diff
@@ -179,18 +251,45 @@ class AmbiguousResult(Exception):
     class NoResult(Exception):
         """Custom Exception for No Query Result"""
 
-    def __init__(self):
+    class NoConnection(Exception):
+        """To be raised if there is no internet connection and no cached result."""
+
+
+    @staticmethod
+    def _is_online(url):
+        host = [x for x in url.split('/')
+                if x not in ['https:', 'http:', '']][0]
+        port = 80 if host.startswith('http://') else 443
+        socket.setdefaulttimeout(1)
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as skt:
+            try:
+                skt.connect((host, port))
+                return True
+            except socket.error:
```

Comment:
will this distinguish between errors from 'no internet connection' and 'the data API is down'?
and do we want to?

Reply:
No, not really. Response time may be different. And no, I don't think we want to.